site/app/controllers/student/TeamController.php

```diff
@@ -165,14 +165,16 @@ public function sendInvitation($gradeable_id) {
             $this->core->redirect($return_url);
         }
 
-        $invite_id = $_POST['invite_id'];
-        if ($this->core->getQueries()->getUserByID($invite_id) === null) {
+        $invite_id = self::cleanInviteId($_POST['invite_id']);
+        $invited_user = $this->core->getQueries()->getUserById($invite_id);
+
+        if (empty($invited_user)) {
```

Comment:
You shouldn't use `empty` on php strings because of its weird type casting rules e.g the string "0" is considered an empty string when its technically a valid user_id

a stronger comparison would be 
```php
if ( $string === '') {
//handle empty string
```

Reply:
Happy to change to null comparison (I assume you were referring to the `empty($invited_user)`), taking a second look before I do:
```php
// site/app/libraries/database/DatabaseQueries.php
143     /**
144      * Gets a user from the database given a user_id.
145      */
146     public function getUserById(string $user_id): ?User {
147         return $this->getUser($user_id);
148     }
```
Since `getUserById` is returning a nullable User, it should either be User or null that should allow us to use `empty($invited_user)` right?

Reply:
I would still compare the object to null php would convert the object into an array before performing the empty check and I don't trust php to perform any implicit cat

It's a bit more clear too, you can use `isset` as well for $user I think that's used more often though submitty

Reply:
👍 will both to `isset`